webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/DynGroupService.scala

```diff
@@ -106,6 +123,61 @@ class DynGroupServiceImpl(
       dyngroupIds.sortBy(numberOfQuery)
     }
   }.toBox
+
+  override def changesSince(lastTime: DateTime): Box[Boolean] = {
+    val n0 = System.currentTimeMillis
+    if(n0 < lastTime.getMillis || n0 - lastTime.getMillis < 100) {
+      Full(true)
+    } else {
+      /*
+       * We want to see if an entry in:
+       * - ou=inventories
+       *   - DN: ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration
+       *   - DN: ou=Removed Inventories,ou=Inventories,cn=rudder-configuration
+       *   - DN: ou=Software,ou=Inventories,cn=rudder-configuration
+       * - ou=nodes: DN: ou=Nodes,cn=rudder-configuration (it will also trigger if compliance mode etc change, but no way to filter out that)
+       * - ou=groups: DN: groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration (we can filter out categories here)
+       */
+      val searchRequest = new SearchRequest("cn=rudder-configuration", Sub.toUnboundid, DereferencePolicy.NEVER, 1, 0, false
+        , AND(
+              OR(
+                  // ou=Removed Inventories,ou=Inventories,cn=rudder-configuration
+                  Filter.create(s"entryDN:dnSubtreeMatch:=ou=Removed Inventories,ou=Inventories,cn=rudder-configuration")
+                  // ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration
+                , Filter.create(s"entryDN:dnSubtreeMatch:=ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration")
+                , AND(IS(OC_SOFTWARE), Filter.create(s"entryDN:dnSubtreeMatch:=ou=Software,ou=Inventories,cn=rudder-configuration"))
+                  // ou=Nodes,cn=rudder-configuration - the objectClass is used only here
+                , AND(IS(OC_RUDDER_NODE), Filter.create(s"entryDN:dnOneLevelMatch:=ou=Nodes,cn=rudder-configuration"))
+                , AND(IS(OC_RUDDER_NODE_GROUP), Filter.create(s"entryDN:dnSubtreeMatch:=groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration"))
+              )
+            , GTEQ("modifyTimestamp", GeneralizedTime(lastTime).toString)
```

Comment:
we don't have index for that. It will be really costly on ldap

Reply:
also, is there a way to stop the query if one result is found ?

Reply:
it's actually stop on the first (or second?) hit, so it is (a bit, but still much less that group search) costly only if there's no modifications. We do something very similar for cache, and it's quite fast. 
(openldap seem to be very efficient in pruning entries that he don't care on that. The fact that keys are sorted in mdb may be the reason, I don't know).